coalib/bearlib/languages/documentation/DocBaseClass.py

```diff
@@ -43,9 +45,17 @@ def extract(content, language, docstyle):
                                    DocumentationComment found in the content.
         """
         docstyle_definition = DocstyleDefinition.load(language, docstyle)
-        return extract_documentation_with_markers(
+        doc_comment_iterator = extract_documentation_with_markers(
                         content, docstyle_definition)
 
+        doc_comment_iterator = DocBaseClass.instantiate_padding(
+            content, doc_comment_iterator)
+
+        doc_comment_iterator = DocBaseClass.instantiate_docstring_type(
+            content, doc_comment_iterator)
```

Comment:
This is still unchanged and very inefficient. This method should be generator.

You have two options: 

1) Change `instantiate_padding` and `instantiate_docstring_type` to only process one docstring at a time, so this can become a generator.
2) move the functionality from `instantiate_padding` and `instantiate_docstring_type` into `extract_documentation_with_markers` and use `yield from`

I prefer the second option.